### PR TITLE
VUU52: Add E2E tests to CI

### LINKED ISSUE #52: Cypress to run in pipeline
### Background
We want the e2e automated test to be run across different browsers as part of our checks whenever a PR is raised

### Work Required
- add yml file (if not there yet)
- add jobs to the yml for running tests cross browser: chrome, firefox and edge
- add action in GH with trigger being PR raised

### Useful Links
https://docs.cypress.io/guides/guides/cross-browser-testing

### PR DESCRIPTION
**Pull request by pling-scottlogic**
### Description
Adds a job to the Test stage of the CI/CD pipeline to run Cypress end-to-end tests whenever a PR is raised. Tests are set to run in Chrome and Edge until the Firefox bug is fixed (#30).

### Change List
- Add a job to `test-ui.yml` to run Cypress E2E tests
- Fix screenshot test:
  - Change selector for save layout panel
  - Update `SHELL_WITH_NEW_THEME_URL` constant to match changes to the `NewTheme` example
  - Point Cypress config to the correct index HTML file location

Closes #52

### CODE REVIEW

**Review by cfisher-scottlogic — APPROVED** [review]
LGTM!

I think it could be worth exploring screenshots for failing tests (if this doesn't happen automatically?) So if a test fails, you get a screenshot of the page to see what's going on which helps debug things (how far did it get, what can the user see, etc). I don't think it's essential because we can just do that locally if this action fails, but if it's not a difficult thing to achieve it might be worth it.

File: .github/workflows/test-ui.yml [hunk 1]
```diff
@@ -23,6 +23,45 @@ jobs:
         run: cd ./vuu-ui && npm install
       - run: cd ./vuu-ui && npm run test:vite
 
+  cypress-e2e:
+    # As a third party action, cypress-io is pinned to a full length commit SHA for security purposes.
+    # This is also a requirement for the semgrep (static code analysis) scan to pass.
+    # https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions#using-third-party-actions
+    runs-on: ubuntu-latest
```
Comment (vferraro-scottlogic) [inline .github/workflows/test-ui.yml:30]: did you consider using windows-latest instead of ubuntu?

File: .github/workflows/test-ui.yml [hunk 2]
```diff
@@ -23,6 +23,45 @@ jobs:
...
+        uses: actions/checkout@v3
+      - name: Use Node.js
+        uses: actions/setup-node@v3
+        with:
+          node-version: "16"
+      - name: Cache dependencies
+        uses: actions/cache@v2
+        with:
+          path: ~/.npm
+          key: npm-${{ hashFiles('package-lock.json') }}
```
Comment (vferraro-scottlogic) [inline .github/workflows/test-ui.yml:42]: not sure I understand what this line does

File: .github/workflows/test-ui.yml (hunk 1 above)
Comment (pling-scottlogic) [reply .github/workflows/test-ui.yml:30]: I went with Ubuntu because that's what all the other actions are using. I guess it could be argued that e2e tests should match an average end-user's environment as close as possible. I don't know if there would be any implication on provisioning different runners though.

**Review by vferraro-scottlogic — APPROVED** [review]
